### ISSUE TITLE
Currency/cost change isn't reflected in Power Total Cost

### ISSUE THREAD

**Issue by Stooovie** [home-assistant/frontend#9928]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

After changing currency in HA Settings - General to reflect my actual currency (CZK) and adjusting cost of energy to that currency, the Total Cost does not get recalculated and shows the original value.

Old currency = EUR
Old cost/kwh= 0.17 eur/kwh
Old total = 2.19
New currency = CZK
New cost/kwh = 4.4 CZK/kwh
**New total = 2.19 (same as before, which is wrong. It should be 56,8)**

### Describe the behavior you expected

Expected the value of Cost gets recalculated to the new currency and cost.

### Steps to reproduce the issue

1. have Energy UI show total cost in currency A (say, EUR)
2. change currency in HA Settings - General
3. change energy cost in Energy settings
...

### What version of Home Assistant Core has the issue?

2021.9.0

**Comment by bramkragten** [comment]
We can't do this, if your energy price actually changes, because you have a new contract or variable prices, it would only be for the future energy you have used. The previous energy would still be charged for the old price.

**Comment by Stooovie** [comment]
But that's not the only thing. After I changed the currency, it simply doesn't reflect the cost. I have it at 4,40 CZK/kWh, I have 1.18 kWh yesterday and total cost shows 0.00.

**Comment by bramkragten** [comment]
Yeah, statistics can't cope with a change of UOM, you would have to clear the statistics for the cost sensor. I'm closing this issues as this is not a frontend issue.

**Comment by Stooovie** [comment]
Can you please tell me how to do that? I have searched the Forum but I'm none the wiser. There should be a reset button somewhere in the UI. Thanks!

**Comment by bramkragten** [comment]
In your database do this: (adjust the entity id of your cost sensor)

```sql
UPDATE statistics_meta SET unit_of_measurement = "CZK" WHERE statistic_id = "sensor.energy_from_grid_cost";
```

To clear the old data, do:
```sql
DELETE FROM statistics_meta WHERE statistic_id = "sensor.energy_from_grid_cost";
```

**Comment by Stooovie** [comment]
Turns out I don't know how to edit a MariaDB database. Thanks anyway.

**Comment by bramkragten** [comment]
There are a lot of tools to connect to a mariaDB and run queries, and browse the data.

https://mariadb.com/kb/en/graphical-and-enhanced-clients/

**Comment by Stooovie** [comment]
Thank you, that's too much for me. I can't deal with manually editing databases to reset a stat.

**Comment by Stooovie** [comment]
Hi, I eventually did edit the database with MyPhpAdmin but the issue still stands. Cost is always stuck at zero. 
![9FBBB1B8-F627-4561-937B-CB45636E1DED](https://user-images.githubusercontent.com/10572373/132137411-a2e19632-84aa-4c05-be4f-603ae36fe74c.png)

**Comment by matlimatli** [comment]
@Stooovie Did you solve this eventually? 

It seems I have a similar problem. In my case the main currency was set to EUR (from default, I suppose, because I have had not reason to ever modify it previously) when the Energy component created the cost/compensation sensors. The actual values collected was however in my home currency (SEK), but the unit was displayed wrongly as EUR. Therefore I changed the main currency to SEK. Now the statistics constantly show 0,00. 

I updated the statistics_meta table to have unit_of_measurement='SEK', as this corresponds to what is actually stored in the statistics table. But I still only get 0,00 in the cost column.
